ansible/configs/ocp-clientvm-rhel8/default_vars_ec2.yml

```diff
@@ -0,0 +1,91 @@
+### AWS EC2 Environment settings
```

Comment:
I'm not sure why we don't have `cloud_provider` set in here. I thought it was supposed to be set and I've seen it set to both `ec2` and `aws`. Should it be set and which is the correct one?

Reply:
Because `cloud_provider`, along with `env_type` must always be passed as extra vars. They are the top-level / mandatory / static variables.

If you look at `ansible/main.yml` you'll see things like

```yaml
- import_playbook: "./configs/{{ env_type }}/pre_infra.yml"
```

And since `import_playbook` is static and imported and "compiled" in the early stages of runtime you cannot define `env_type` in a file that is dynamically included with an `include_vars` task.

Also, `cloud_provider` must be defined before the file you're mentioning is included: `default_vars_`**`ec2`**`.yml`.  That's dynamically included via `include_vars` and `default_vars_{{ cloud_provider }}.yml`.

Reply:
So my question was...why don't we have `cloud_provider` defined in this `default_vars_ec2` file? And how does this work without it? Because it does work. And why is it `ec2` instead of `aws`?

Reply:
Nevermind me on why does it work...it was in my vars file that I pass in.

Back to my original question, should we have `cloud_provider: ec2` set as a default in this file ?

Reply:
`cloud_provider` and `env_type` are always passed as extra vars.

So to reply to your question:
> Back to my original question, should we have cloud_provider: ec2 set as a default in this file ?

No, it would not have any effect and would create confusion i think.

Reply:
It would create confusion to have the `cloud_provider` set in the default vars file for the...cloud provider? I would think that makes it more user friendly so it is one less thing a user has to feed in as an extra var.